connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -107,15 +107,17 @@ message Join {
   }
 }
 
-// Relation of type [[Union]], at least one input must be set.
```

Comment:
This documentation (`at least one input must be set`) does not match the implementation `assert(u.getInputsCount == 2, "Union must have 2 inputs")`. To avoid confusion I choose to remove this for now.

Once our implementation is stable, we can go back to re-document the semantic.

Reply:
TBH i think the comment was not correct in the first place. I mean while a set can have one element, the question is does it make sense to have a set plan with only one?

Reply:
I am thinking it should be always two inputs so the implementation should be correct. Though I don't know enough about dataframe API usage so not sure if there could be only one, or more than two, etc.

Given proto is also a API layer, before 3.4 release I think we need to add a lot more comments on proto to clarify each one's semantic. We can finalize how many inputs are possible at that time.